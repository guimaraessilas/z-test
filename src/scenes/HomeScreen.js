import React, { useState } from "react";
import {
  View,
  Text,
  StyleSheet,
  TextInput,
  TouchableHighlight,
  Alert,
  ActivityIndicator,
} from "react-native";
import Header from "../components/Header";
import { useLazyQuery } from "react-apollo";
import gql from "graphql-tag";
import { getLocation } from "../utils/maps-service";

const GET_PLACES = gql`
  query pocSearchMethod(
    $now: DateTime!
    $algorithm: String!
    $lat: String!
    $long: String!
  ) {
    pocSearch(now: $now, algorithm: $algorithm, lat: $lat, long: $long) {
      id
    }
  }
`;

const HomeScreen = ({ navigation }) => {
  const [address, setAddress] = useState("");
  const [loadStore, { called, loading, data }] = useLazyQuery(GET_PLACES);

  function handleSearch() {
    if (address) {
      getLocation(address).then((res) => {
        loadStore({
          variables: {
            algorithm: "NEAREST",
            lat: JSON.parse(res).lat.toString(),
            long: JSON.parse(res).lng.toString(),
            now: new Date().toISOString(),
          },
        });
      });
    } else {
      Alert.alert("Atenção", "Você esqueceu de digitar o seu endereço...");
    }
  }

  if (data) {
    navigation.navigate("Product", { storeData: data.pocSearch });
  }

  return (
    <View style={styles.container}>
      <Header title={"Zé Cachaceiro"} />
      <View style={styles.content}>
        <Text style={styles.title}>Hey, onde vamos beber hoje?</Text>
        <TextInput
          placeholder="Digite seu endereço aqui"
          style={styles.inputAddress}
          onChangeText={(addr) => setAddress(addr)}
        />
        <TouchableHighlight
          style={styles.searchButton}
          onPress={() => handleSearch()}
          disabled={loading}
        >
          {loading ? (
            <ActivityIndicator size="small" color="blue" />
          ) : (
            <Text style={styles.textButton}>Buscar</Text>
          )}
        </TouchableHighlight>
        {/** //TODO: ADICIONAR RODAPÉ COM VALOR TOTAL DO CARRINHO */}
      </View>
    </View>
  );
};
export default HomeScreen;

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  content: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    padding: 10,
  },
  title: {
    fontSize: 26,
    textAlign: "center",
  },
  inputAddress: {
    backgroundColor: "white",
    borderRadius: 10,
    borderColor: "#0055cc",
    borderWidth: 2,
    width: "100%",
    height: 60,
    padding: 5,
    paddingLeft: 15,
    marginVertical: 10,
  },
  searchButton: {
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "white",
    height: 60,
    width: 200,
    borderRadius: 10,
  },
  textButton: {
    fontSize: 26,
    color: "#0055cc",
    fontWeight: "600",
  },
  footer: {
    alignItems: "center",
    justifyContent: "center",
    height: 80,
    backgroundColor: "green",
  },
});
