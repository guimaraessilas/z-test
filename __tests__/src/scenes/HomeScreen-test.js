import React from "react";
import { act, create } from "react-test-renderer";
import { useDispatch } from "react-redux";
import HomeScreen from "../../../src/scenes/HomeScreen";

jest.mock("react-redux");

const setup = () => create(<HomeScreen />);

describe("Should test HomeScreen ", () => {
  beforeAll(() => {
    useDispatch.mockImplementation(() => jest.fn());
  });

  describe("snapshot", () => {
    it("render correctly", () => {
      expect(setup().toJSON()).toMatchSnapshot();
    });
  });
});
