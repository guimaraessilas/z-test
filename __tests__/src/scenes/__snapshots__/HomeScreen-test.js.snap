// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test HomeScreen  snapshot render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#0088cc",
        "height": 60,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 26,
          "fontWeight": "700",
        }
      }
    >
      Zeca Chaceiro
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 26,
          "textAlign": "center",
        }
      }
    >
      Hey, onde vamos beber hoje?
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Digite seu endereÃ§o aqui"
      rejectResponderTermination={true}
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "#0055cc",
          "borderRadius": 10,
          "borderWidth": 2,
          "height": 60,
          "marginVertical": 10,
          "padding": 5,
          "paddingLeft": 15,
          "width": "100%",
        }
      }
      underlineColorAndroid="transparent"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderRadius": 10,
          "height": 60,
          "justifyContent": "center",
          "width": 200,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#0055cc",
            "fontSize": 26,
            "fontWeight": "600",
          }
        }
      >
        Buscar
      </Text>
    </View>
  </View>
</View>
`;
